portfolio/src/main/webapp/script.js

```diff
@@ -56,20 +59,29 @@ function createListElement(text) {
 }
 
 /**
- * Fetches user status from servlet and generates html to display.
+ * Handles showing of buttons depending on the fetched status of the user.
+ * Assigns the corresponding URL to the modal buttons (login or logout, depending).
  */
 function checkLoginStatus() {
-  fetch("/user").then(response => response.json()).then((status) => {
-    if(status == "logged") {
-      const logoutBtn = document.getElementById("logout-btn");
+  fetch("/user").then(response => response.json()).then((user) => {
+    if (user.loginStatus) {
+      // Shows the log out button (in navbar and modal) and the form if the user is logged in. 
       const commentForm = document.getElementById("comment-form");
-      logoutBtn.style.display = "block";
+      const logoutBtn = document.getElementById("logout-btn");
+      const logoutModalBtn = document.getElementById("logout-modal-btn"); 
       commentForm.style.display = "block";
+      logoutBtn.style.display = "block";
+      logoutModalBtn.setAttribute("href", user.url);
+      logoutModalBtn.style.display = "block";
```

Comment:
Optional: If you 'return' at the end here, you can remove the 'else' condition and unindent the rest of the code. Up to you - in this case I think the if...else... is good readability, but just something to think about for future if...else... blocks!

Reply:
I'll keep it in mind!